compiler/rustc_typeck/src/coherence/inherent_impls.rs

```diff
@@ -197,6 +197,9 @@ impl<'tcx> InherentCollect<'tcx> {
             ty::Dynamic(data, ..) if data.principal_def_id().is_some() => {
                 self.check_def_id(item, self_ty, data.principal_def_id().unwrap());
             }
+            ty::TyAlias(..) => {
+                span_bug!(ty.span, "unexpected TyAlias in InherentCollect::check_item");
+            }
```

Comment:
Does this get normalized elsewhere, or do we not have a test for:
```rust
struct Foo;
type FooAlias = Foo;
impl FooAlias {}
```

Reply:
It works perfectly fine so it's normalized ahead of this call.